systemtest/src/test/java/io/strimzi/systemtest/report/NodePoolsClusterReportST.java

```diff
@@ -0,0 +1,486 @@
+ */
+package io.strimzi.systemtest.report;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.ReplicaSet;
+import io.fabric8.kubernetes.api.model.networking.v1.NetworkPolicy;
+import io.fabric8.kubernetes.api.model.policy.v1.PodDisruptionBudget;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRole;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
+import io.fabric8.kubernetes.api.model.rbac.Role;
+import io.fabric8.kubernetes.api.model.rbac.RoleBinding;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.api.kafka.model.KafkaBridge;
+import io.strimzi.api.kafka.model.KafkaConnect;
+import io.strimzi.api.kafka.model.KafkaMirrorMaker2;
+import io.strimzi.api.kafka.model.KafkaRebalance;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.KafkaTopic;
+import io.strimzi.api.kafka.model.KafkaUser;
+import io.strimzi.api.kafka.model.StrimziPodSet;
+import io.strimzi.api.kafka.model.nodepool.KafkaNodePool;
+import io.strimzi.api.kafka.model.nodepool.ProcessRoles;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.annotations.ParallelTest;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaBridgeTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaConnectTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaMirrorMaker2Templates;
+import io.strimzi.systemtest.templates.crd.KafkaNodePoolTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaRebalanceTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTopicTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.FileUtils;
+import io.strimzi.test.executor.Exec;
+import io.strimzi.test.k8s.exceptions.KubeClusterException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.TestUtils.USER_PATH;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@Tag(REGRESSION)
+public class NodePoolsClusterReportST extends AbstractClusterReportST {
```

Comment:
Only difference between those two classes are NodePools, so what if we have just one testsuite with the tests, in which we would switch between Kafka with/without NodePools based on `STRIMZI_FEATURE_GATES` env variable?

Yeah it would need few "hacks" and `if`s to make it work, but more or less all helper methods for naming are already in the suite.

The report.sh tests with NodePools would run during KRaft and FG Azure pipelines, the tests for Kafka without NodePools would run during regular pipelines. We will save execution time and also lines of code, but both modes will be tested.

WDYT?

Reply:
In the NodePools implemenation we also have StableConnectIdentities. Some checks are different, and some are missing. 

In general, I don't like the idea of trading code maintenance for infrastructure convenience. Can't we simply tag them differently, so that they can be activated only when a specific pipeline is running?

Reply:
Hmm I see. I know that having one class which will run different checks based on env variables can be confusing.
We can definitely have some annotation to trigger these tests when `NodePools` and `StableConnectIdentities` will be enabled, that's not a problem.

Also to remove "extra lines of code", we can have something similar as for tracing tests -> so in the abstract have methods with checks (something like `doTestCreateReport`) and in the two ST classes just call them? I think that there will be common parts of code, which can be reused + maybe changed for handling both NodePools and non NodePools. (it's just an idea)